<launch>

<arg name="filename" default="$(find akros_behaviors)/config/waypoints_store.yaml"/>
<arg name="localization_frame" default="map"/>
<arg name="base_frame" default="base_link"/>
<arg name="move_base_ns" default="move_base"/>
<arg name="teleop_ns" default="teleop"/>

<!-- Run Waypoint Sequence Store -->
<node pkg="akros_behaviors" type="waypoint_sequence_store.py" name="wp_seq_store" output="screen">
    <remap from="status" to="$(arg teleop_ns)/status"/>
    <remap from="mode" to="$(arg teleop_ns)/mode"/>
    <param name="waypoint_store_yaml" value="$(arg filename)"/>
    <param name="localization_frame" value="$(arg localization_frame)"/>
    <param name="base_frame" value="$(arg base_frame)"/>
</node>

<!-- Run Waypoint Sequence Player -->
<node pkg="akros_behaviors" type="waypoint_sequence_player.py" name="wp_seq_player" output="screen">
    <remap from="mode" to="$(arg teleop_ns)/mode"/>
    <param name="waypoint_store_yaml" value="$(arg filename)"/>
    <param name="move_base_ns" value="$(arg move_base_ns)"/>
    <param name="frame_id" value="$(arg localization_frame)"/>
</node>

</launch>
